vpc/tool/identity/ec2.go

```diff
@@ -74,6 +126,22 @@ func (e *ec2Provider) GetIdentity(ctx context.Context) (*vpcapi.InstanceIdentity
 
 	stats.Record(ctx, getIdentityLatency.M(float64(time.Since(start).Nanoseconds())), getIdentitySuccess.M(1))
 
+	//Write the files to /run
+	iidfile, err := os.OpenFile(InstanceIdentityFile, os.O_WRONLY, 0644)
```

Comment:
Don't ever write files directly that must be written atomically. Use O_TMPFILE + rename. https://github.com/google/renameio